cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml

```diff
@@ -61,14 +61,14 @@ spec:
               then
                 exit 1;
               fi;
-              LAST_MODIFIED_DATE=`stat /var/log/fluentd-buffers | grep Modify | sed -r "s/Modify: (.*)/\1/"`;
-              LAST_MODIFIED_TIMESTAMP=`date -d "$LAST_MODIFIED_DATE" +%s`;
-              if [ `date +%s` -gt `expr $LAST_MODIFIED_TIMESTAMP + $STUCK_THRESHOLD_SECONDS` ];
+              touch -d "@$(date -d"${STUCK_THRESHOLD_SECONDS} seconds ago" '+%s')" /tmp/marker-stuck;
+              if [[ -z "$(find /var/log/fluentd-buffers -type f -newer /tmp/marker-stuck -print -quit)" ]];
```

Comment:
simple `-quit` would indicate success via exit code, so no need to check output:

```
find /var/log/fluentd-buffers -type f -newer /tmp/marker-stuck -quit || {
   rm -rf /var/log/fluentd-buffers;
   exit 1;
}
find /var/log/fluentd-buffers -type f -newer /tmp/marker-liveness -quit || exit 1
```

if you start script with `set -e` , then simple find would be enough in latter case as non successful code would abort script by itself.

`if [ ! -e /var/log/fluentd-buffers ];` check above is redundant, `find` would exit with error if directory is not found

Reply:
According to the [man page](https://linux.die.net/man/1/find), `find` only exits with a non-zero status if errors occur. If all of the files have been processed successfully, the exit status is 0 (in other words, `find` is not `grep` :-) ).

Reply:
I guess "nothing is found" is error condition, because find 100% exits with != 0 if no matches found

Reply:
Something isn't lining up here. `find` only seems to exit 1 when the directory being searched does not exist. If there are no files that match inside of a directory that exists, the exit code is always 0.

```$ ls

$ find . -type f -quit || echo "No files found";

$ touch hello

$ ls
hello

$ find . -type f -quit || echo "No files found";
```

In the example above, "No files found" is never echoed, because the exit code is 0 in both scenarios. I agree that we don't need the `if [ ! -e /var/log/fluentd-buffers ]` condition with the new implementation, but I do need to check if the output of `-print` is empty or not, to truly determine whether or not a file is matched or not.

Reply:
@bmoyles0117 , @igorpeshansky , you were right, just checked on empty dir and it exits 0 indeed.

Reply:
I guess I need to revisit few of my scripts :) thanks!